pandas/_libs/lib.pyx

```diff
@@ -891,9 +891,17 @@ def indices_fast(ndarray index, const int64_t[:] labels, list keys,
     if n == 0:
         return result
 
-    start = 0
-    cur = labels[0]
-    for i in range(1, n):
+    # Start at the first non-null entry
+    j = 0
+    for j in range(0, n):
+        if labels[j] != -1:
+            break
+    else:
+        raise ValueError("cannot handle all null values")
```

Comment:
hmm, just return result no?

Reply:
do any tests hit this?

Reply:
Good call, returning result is much better than raising. Currently no tests can possibly hit this because of short-circuit logic in `get_indexer_dict` (I changed the comment around it, see the diff), I merely added it to be defensive. Doesn't seem advantageous to remove the short-circuit logic.

Reply:
The return of an empty dictionary is hit by several tests however.